# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/tests/test_bug1836801.html
# Commit: fb89eb43b3ce
# Full Hash: fb89eb43b3ce78c5031edb49a2e62ddfa7988415
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2024-02-23 03:40:30
# Description:
#   Bug 1836801. If the order of top layer items changes make sure to mark the frame modified. r=gfx-reviewers,aosmond
#   
#   The order of items in the mTopLayer array determines the order that we build display items, so if it changes we need to mark the frames as modified so that retained display list partial updates can properly merge.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D201214
# ==============================================================================

diff -r c706f3410c62 -r fb89eb43b3ce layout/base/tests/test_bug1836801.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/layout/base/tests/test_bug1836801.html	Thu Feb 22 10:09:18 2024 +0000
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <script src="/tests/SimpleTest/SimpleTest.js"></script>
+  <script src="/tests/SimpleTest/EventUtils.js"></script>
+  <link rel="stylesheet" type="text/css" href="/tests/SimpleTest/test.css"/>
+</head>
+
+<script>
+// this is a crashtest, just getting to the end is a pass, fullscreen
+// doesn't work in crashtests though not sure why -->
+SimpleTest.waitForExplicitFinish();
+
+SimpleTest.requestFlakyTimeout("crashtest sensitive to timing");
+
+function begin() {
+  SimpleTest.waitForFocus(begin2);
+}
+
+function begin2() {
+  SpecialPowers.pushPrefEnv({
+    "set":[["full-screen-api.allow-trusted-requests-only", false]]
+  }, startTest);
+}
+
+window.addEventListener("load", begin);
+
+const func_0 = async function(arg0) {
+  //SpecialPowers.wrap(document).notifyUserGestureActivation();
+  await arg0.originalTarget.requestFullscreen()
+  arg0.originalTarget.ariaValueText = "a"
+}
+
+async function startTest() {
+  let a = document.createElementNS("http://www.w3.org/1999/xhtml", "canvas")
+  document.documentElement.appendChild(a)
+  document.addEventListener("DOMAttrModified", func_0, { })
+  let b = document.createElementNS("http://www.w3.org/1999/xhtml", "canvas")
+  document.documentElement.appendChild(b)
+  b.setAttribute("class", "x")
+  //SpecialPowers.wrap(document).notifyUserGestureActivation();
+  await b.mozRequestFullScreen()
+  let c = document.createElementNS("http://www.w3.org/1999/xhtml", "slot")
+  document.documentElement.appendChild(c)
+  c.setAttribute("class", "x")
+  b.inert = true
+  setTimeout(async () => {
+    b.setAttribute("title", "a")
+  }, 200)
+  a.ariaSort = "descending"
+  setTimeout(finishup, 400);
+}
+async function finishup() {
+  await new Promise(resolve => requestAnimationFrame(() => requestAnimationFrame(resolve)));
+  await document.exitFullscreen();
+  SimpleTest.finish();
+}
+</script>
+</html>
